artemis-server/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/NettyAcceptor.java

```diff
@@ -401,6 +406,21 @@ public ActiveMQThreadFactory run() {
             acceptorType = KQUEUE_ACCEPTOR_TYPE;
 
             logger.debug("Acceptor using native kqueue");
+         } else if (useIoUring && CheckDependencies.isIoUringAvailable()) {
```

Comment:
I would expect io uring to be primary option later with fallback to epoll and then nio, as such maybe this should be first in order of preference. Its off by default but this way going forwards correct preference order would exist.

Reply:
Yes :) correct

Reply:
Damn karaf and its feature checks :O it works for reflection too? :O

Reply:
You can probably add some metadata to artemis-server-osgi let it know the io.netty.incubator.channel.uring package usages are optional.